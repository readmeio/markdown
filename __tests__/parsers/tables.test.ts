import { removePosition } from 'unist-util-remove-position';

import { mdast } from '../../index';

describe('table parser', () => {
  describe('unescaping pipes', () => {
    it('parses tables with pipes in inline code', () => {
      const doc = `
|              |    |
| :----------- | :- |
| \`foo \\| bar\` |    |
`;
      const ast = mdast(doc);
      removePosition(ast, { force: true });

      expect(ast).toMatchSnapshot();
    });

    it('parses tables with pipes', () => {
      const doc = `
|            |    |
| :--------- | :- |
| foo \\| bar |    |
`;
      const ast = mdast(doc);
      removePosition(ast, { force: true });

      expect(ast).toMatchSnapshot();
    });

    it('parses jsx tables with pipes in inline code', () => {
      const doc = `
<Table align={["left","left"]}>
  <thead>
    <tr>
      <th style={{ textAlign: "left" }}>
        force
        jsx
      </th>

      <th style={{ textAlign: "left" }}>

      </th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td style={{ textAlign: "left" }}>
        \`foo | bar\`
      </td>

      <td style={{ textAlign: "left" }}>

      </td>
    </tr>
  </tbody>
</Table>
`;

      const ast = mdast(doc);
      removePosition(ast, { force: true });

      expect(ast).toMatchSnapshot();
    });
  });
});
