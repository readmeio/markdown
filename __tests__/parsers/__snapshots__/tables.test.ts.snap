// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`table parser > unescaping pipes > parses jsx tables with pipes in inline code 1`] = `
{
  "children": [
    {
      "align": [
        "left",
        "left",
      ],
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "children": [
                    {
                      "type": "text",
                      "value": "force
jsx",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        {
          "children": [
            {
              "children": [
                {
                  "children": [
                    {
                      "type": "inlineCode",
                      "value": "foo | bar",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "tableau",
    },
  ],
  "type": "root",
}
`;

exports[`table parser > unescaping pipes > parses tables with pipes 1`] = `
{
  "children": [
    {
      "align": [
        "left",
        "left",
      ],
      "children": [
        {
          "children": [
            {
              "children": [],
              "type": "tableCell",
            },
            {
              "children": [],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        {
          "children": [
            {
              "children": [
                {
                  "type": "text",
                  "value": "foo | bar",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "root",
}
`;

exports[`table parser > unescaping pipes > parses tables with pipes in inline code 1`] = `
{
  "children": [
    {
      "align": [
        "left",
        "left",
      ],
      "children": [
        {
          "children": [
            {
              "children": [],
              "type": "tableCell",
            },
            {
              "children": [],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        {
          "children": [
            {
              "children": [
                {
                  "type": "inlineCode",
                  "value": "foo | bar",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "root",
}
`;
